### ISSUE TITLE
Metrics/LineLength should ignore rubocop comments

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#6964]
I believe the `Metrics/LineLength` cop should ignore `# rubocop: disable` comments, or at least those that end with `# rubocop: disable Metrics/LineLength`.

Otherwise we get a catch-22 where the "solutions" are all worse than the "problems"

Example:

```ruby
          fail_due_to 'multiple variable patterns in same sequence' if @variadic_index && !@arities.one?(Range)
# => Metrics/LineLength violation (and indeed the line is overly long and complex)

          many_variadic = variadic_index && something_else
          fail_due_to 'multiple variable patterns in same sequence' if many_variadic
# => still a Metrics/LineLength violation

          if variadic_index && something_else
            fail_due_to 'multiple variable patterns in same sequence'
          end
# => now that's a Style/GuardClause violation

          if variadic_index && something_else # rubocop:disable Style/GuardClause
            fail_due_to 'multiple variable patterns in same sequence'
          end
# => We're back to a Metrics/LineLength violation because of the disable comment

          if variadic_index && something_else # rubocop:disable Style/GuardClause Metrics/LineLength
            fail_due_to 'multiple variable patterns in same sequence'
          end
# => Still a Metrics/LineLength violation

# Supposedly "better":

          # Use less descriptive variable or method name:
          many_var = variadic_index && something_else
          fail_due_to 'multiple variable patterns in same sequence' if many_var

          # Make disabling comment multiline, although it
          # really is only for a single line

          # Line break 
          fail_due_to 'multiple variable patterns in same sequence' \
            if variadic_index && something_else

          # etc.
```

**Comment by deivid-rodriguez** [comment]
There's a `IgnoreCopDirectives` configuration for this cop that does just this.

**Comment by marcandre** [comment]
I'm personally not using this cop, just fighting with it when playing with rubocop's own code. 

I think rubocop is using it's own defaults, right? Maybe the default could be changed?

**Comment by deivid-rodriguez** [comment]
Me neither, I never use this cop. I just saw the configuration by chance while playing around with rubocop's source and thought it could be useful here.

> I think rubocop is using it's own defaults, right? Maybe the default could be changed?

Not exactly, it tweaks them a bit: https://github.com/rubocop-hq/rubocop/blob/master/.rubocop.yml, but yeah, the default should probably be changed since I'm guessing this problem affects most people using this cop?

**Comment by marcandre** [comment]
Bump
